ee/codegen/src/__test__/helpers/node-data-factories.ts

```diff
@@ -1153,7 +1158,9 @@ export function apiNodeFactory({
       methodInputId: "9bf086d4-feed-47ff-9736-a5a6aa3a11cc",
       urlInputId: "480a4c12-22d6-4223-a38a-85db5eda118c",
       bodyInputId: "74865eb7-cdaf-4d40-a499-0a6505e72680",
-      authorizationTypeInputId: "de330dac-05b1-4e78-bee7-7452203af3d5",
+      authorizationTypeInputId: useUndefinedAuthorizationTypeInputId
+        ? undefined
+        : defaultAuthorizationTypeInputId,
```

Comment:
[nit] even if we do this, the `authorizationTypeInput` is in `node_inputs`. my preference would be to define the following on the factory:

```typescript
authorizationTypeInput?: NodeInput
```

then we could use `authorizationTypeInput.id` here if present and add it to the `inputs` list below if present

Reply:
could use `undefined` vs `null` to disambiguate between using a default value so that we don't have to update a bunch of tests and a test opting into "ignore this field"